.travis.yml

```diff
@@ -27,11 +27,26 @@ install:
 # command to run tests
 script:
   - ./testsuite/MDAnalysisTests/mda_nosetests -v --with-coverage --cover-package MDAnalysis --processes=2 --process-timeout=120 --with-memleak
-  - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_BRANCH == "develop" && cd ./package && python setup.py build_ext --inplace && cd ./doc/sphinx && make clean html || true
+  - |
+     test ${TRAVIS_PULL_REQUEST} == "false" && \
+     test ${TRAVIS_BRANCH} == ${GH_DOC_BRANCH} && \
+     test "${TRAVIS_BUILD_NUMBER}.1" == "${TRAVIS_JOB_NUMBER}" && \
+     (cd ./package && python setup.py build_ext --inplace) && \
+     (cd ./package/doc/sphinx && make clean html) || \
+     true
```

Comment:
Note that we currently always report a success from the building docs stage.

At some point we can consider failing the job if the docs don't build properly (just remove the `|| true`).